src/components/ThirdPartyCollectionDetailPage/ThirdPartyCollectionDetailPage.tsx

```diff
@@ -48,6 +48,19 @@ export default class ThirdPartyCollectionDetailPage extends React.PureComponent<
     isAuthModalOpen: false
   }
 
+  componentDidUpdate() {
+    const { collection, itemCurations, onFetchItemCurations, isLoadingItemCurations, itemCurationsError } = this.props
+    if (
```

Comment:
I'm not sold on this implementation but I'm not convinced on moving the fetching logic to a sagas neither (like one listening to the route changes). 
@LautaroPetaccio @nicosantangelo Any thoughts about where to put this logic guys?

Reply:
tbh I think this is the correct place to do this. The improvement I can come up with is just doing

`if(this.shouldFetchCurations())`

and hide the condition there

Reply:
good idea! Updated to something similar to that :)